modules/webgl/test/classes/texture.spec.js

```diff
@@ -574,3 +574,44 @@ test('WebGL2#Texture2D setSubImageData', t => {
 
   t.end();
 });
+
+test('WebGL2#Texture2D resize', t => {
+  const {gl} = fixture;
+  let texture = new Texture2D(gl, {
+    data: null,
+    width: 2,
+    height: 2,
+    mipmaps: true
+  });
+
+  texture.resize({
+    width: 4,
+    height: 4
+  });
+
+  t.ok(texture.mipmaps, 'mipmaps should be the same as before resizing for POT texture.');
+
+  texture = new Texture2D(gl, {
+    data: null,
+    width: 2,
+    height: 2,
+    mipmaps: false
+  });
+
+  texture.resize({
+    width: 4,
+    height: 4
+  });
+
+  t.notOk(texture.mipmaps, 'mipmaps should be the same as before resizing for POT texture.');
+
+  texture.resize({
```

Comment:
I will move this to line 594, so we are testing mipmaps: true -> resize to NPOT -> verify mipmaps is false.